controllers/nodehealthcheck_controller.go

```diff
@@ -204,8 +208,28 @@ func allNHCHandler(c client.Client) handler.MapFunc {
 			return nil
 		}
 		var r []reconcile.Request
-		for _, n := range nhcList.Items {
-			r = append(r, reconcile.Request{NamespacedName: types.NamespacedName{Name: n.GetName()}})
+		for _, nhc := range nhcList.Items {
+			nodes := v1.NodeList{}
+			selector, err := metav1.LabelSelectorAsSelector(&nhc.Spec.Selector)
+			if err != nil {
+				logger.Error(err, "failed to use the NHC selector. Adding the NHC to the reconcile queue anyway")
+			} else {
+				_ = c.List(context.Background(), &nodes, &client.ListOptions{
+					LabelSelector: selector,
+				})
+				var found bool
+				for _, node := range nodes.Items {
+					if node.GetName() == o.GetName() {
+						found = true
+						break
+					}
+				}
+				if !found {
+					break
+				}
```

Comment:
why this is needed?
I think this means we'll only scan the first NHC CR in the list, if there was no match in it.

maybe you meant to use `continue`?

Reply:
I simplified it now